test/exe/version_out_test.sh

```diff
@@ -2,6 +2,9 @@
 
 set -e -o pipefail
 
+# Undo LC_ALL=en_US.UTF-8, since it breaks sed.
+export LC_ALL=C
```

Comment:
how does it break sed ?

Reply:
i can see https://www.gnu.org/software/sed/manual/html_node/Locale-Considerations.html

but what is being seded here (afaict) is the version string which shouldnt contain any multibyte chars - which would imply the issue is more with the sed string - i might be wrong

Reply:
```
==================== Test output for //test/exe:version_out_test:
sed: -e expression #1, char 127: Invalid range end
```

See failed run: https://dev.azure.com/cncf/envoy/_build/results?buildId=71987&view=logs&j=5b1da4ae-91c5-53b4-2dce-4cdfc8f90e84&t=7ce850fe-192d-5106-1bd3-aa6c6dba22dc

Reply:
not sure - but afaict both the linux image and the build-tools can be updated without this change

it may be necessary for some other reason that im not understanding - but my reading of the source is that its using sed to change the verssion string - and that should not have any kind of utf-8 issues

Reply:
re this @PiotrSikora - i think - and i may be wrong - that the sed expression has an incorrect `\` which is being interpreted as a utf-8 multibyte char but i really dont see how the change in the docker image recipe would affect this

Reply:
ie its unrelated to the change in envoy-build-tools - it was always there - and its only when one of the tests fails that it has this issue

Reply:
I mean, what can I say? See the CI failures from a few hours ago.

Also, the fact that updating Linux image alone works, and updating build-tools alone works, doesn't prove that both changes work fine together, does it? It's not like I was making all those changes for fun, they are direct results of failures on the CI.